x/wasm/keeper/keeper_mpt.go

```diff
@@ -0,0 +1,61 @@
+package keeper
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	apptypes "github.com/okx/okbchain/app/types"
+	clientcontext "github.com/okx/okbchain/libs/cosmos-sdk/client/context"
+	"github.com/okx/okbchain/libs/cosmos-sdk/store/mpt"
+	sdk "github.com/okx/okbchain/libs/cosmos-sdk/types"
+	"github.com/okx/okbchain/libs/cosmos-sdk/x/auth"
+	"github.com/okx/okbchain/x/wasm/watcher"
+	"log"
+)
+
+func (k Keeper) getStorageStore(ctx sdk.Context, acc sdk.WasmAddress) sdk.KVStore {
+	account := k.accountKeeper.GetAccount(ctx, sdk.WasmToAccAddress(acc))
+	if account == nil && watcher.Enable() {
+		account = getAccount(acc)
+	}
+	ethAcc := common.BytesToAddress(acc.Bytes())
+
+	// in case of query panic
+	var stateRoot common.Hash
+	if account == nil {
+		stateRoot = ethtypes.EmptyRootHash
+	} else {
+		stateRoot = account.GetStateRoot()
+	}
+	return k.ada.NewStore(ctx.GasMeter(), ctx.KVStore(k.storageStoreKey), mpt.AddressStoragePrefixMpt(ethAcc, stateRoot))
+}
+
+func (k Keeper) GetStorageStore4Query(ctx sdk.Context, acc sdk.WasmAddress) sdk.KVStore {
+	return k.getStorageStore(ctx, acc)
+}
+
+var clientCtx clientcontext.CLIContext
+
+func SetCliContext(ctx clientcontext.CLIContext) {
+	clientCtx = ctx
+}
+
+func getAccount(addr sdk.WasmAddress) *apptypes.EthAccount {
```

Comment:
wasm watchdb has a bug now, so it is not necessary after bug fix of wasm watchdb